src/frontend/screens/Audio/useAudioPlayback.ts

```diff
@@ -31,27 +31,31 @@ export const useAudioPlayback = (recordingUri: string) => {
   }, []);
 
   useEffect(() => {
-    let soundInstance: Sound | null = null;
+    let isCancelled = false;
+
     Audio.Sound.createAsync({uri: recordingUri})
       .then(({sound, status}) => {
-        if ('error' in status && status.error) {
-          setError(new Error(status.error));
+        if (isCancelled) {
+          sound.unloadAsync().catch(() => {});
           return;
         }
-        soundInstance = sound;
         recordedSoundRef.current = sound;
         setDuration((status as AVPlaybackStatusSuccess).durationMillis ?? 0);
         sound.setOnPlaybackStatusUpdate(audioCallbackHandler);
       })
       .catch(err => {
-        handleError(err);
+        if (!isCancelled) {
+          handleError(err);
+        }
       });
 
     return () => {
-      if (soundInstance) {
-        soundInstance.unloadAsync().catch(err => {
+      isCancelled = true;
+      if (recordedSoundRef.current) {
+        recordedSoundRef.current.unloadAsync().catch(err => {
           handleError(err);
```

Comment:
why are we handling the error here but not on line 29. I think in both cases we should skip the error as the app will still function if the sound isn't unloaded